core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -239,6 +242,53 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
   }
 
+  val unresolvedStarRegex = """(cannot resolve ')(\w+).(\w+).*(' give input columns.*)""".r
```

Comment:
Move all this code to some utility class.

Reply:
@rishitesh -- Since this is very specific to this fix and we would certainly want to revisit so I kept it right there. I didn't want to have references to this fix in any other part of the code. In fact I added these regexp in the method itself but then like for every analysis exception I didn't want to create the regexp object.

Reply:
Just don't want SnappySession to get riddled with specific handling. But yes if we are revisiting then its fine.

Reply:
Ok. Let me file a task for this.

Reply:
Can move the regex to SnappySession object so that it is singleton and does not get created for every session.

Reply:
Ok. Will do. But waiting for the response on the main implementation approach. In the parser if I move the logic then this regexp will anyways not be required.